### DISCUSSION TITLE
iobroker startet nicht bei volume binding

### DISCUSSION THREAD

**Discussion by IrgendwerUsingGitHub** [buanet/ioBroker.docker#321] [Questions & Answers / Fragen & Antworten] [answered]
Hallo,

ich hatte die letzten Tage viele Probleme mit iobroker im docker. Es hat immer wunderbar funktioniert, aber seit ein paar Tagen ist der Wurm drin.
Ich glaube das Problem runtergebrochen zu haben auf das binding von den data Ordnern. Wenn ich "docker create volume" benutze funktioniert es.
Ich habe versucht dem binding Ordner und sämtlichen Inhalt chmod 777 zu setzen, aber leider erfolglos.

Leider geben alle Logs die ich gefunden habe nicht viel her.

```
--------------------------------------------------------------------------------
-----                          System Information                          -----
-----                    arch:                armv7l                       -----
-----                    hostname:            5b16d513a98b                 -----
-----                                                                      -----
-----                          Version Information                         -----
-----                    image:               v7.2.0                       -----
-----                    build:               2023-02-10T23:45:26+00:00    -----
-----                    node:                v16.19.0                     -----
-----                    npm:                 8.19.3                       -----
-----                                                                      -----
-----                        Environment Variables                         -----
-----                    DEBUG:               true                         -----
-----                    SETGID:              1000                         -----
-----                    SETUID:              1000                         -----
--------------------------------------------------------------------------------
 
--------------------------------------------------------------------------------
-----                  Step 1 of 5: Preparing container                   -----
--------------------------------------------------------------------------------
 
Updating Linux packages on first run... 
W: Failed to fetch http://deb.debian.org/debian/dists/bullseye/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://deb.debian.org/debian-security/dists/bullseye-security/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://deb.debian.org/debian/dists/bullseye-updates/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch https://deb.nodesource.com/node_16.x/dists/bullseye/InRelease  Temporary failure resolving 'deb.nodesource.com'
W: Some index files failed to download. They have been ignored, or old ones used instead.
Done.
 
Registering maintenance script as command... Done.
 
--------------------------------------------------------------------------------
-----             Step 2 of 5: Detecting ioBroker installation             -----
--------------------------------------------------------------------------------
 
Existing installation of ioBroker detected in "/opt/iobroker".
 
--------------------------------------------------------------------------------
-----             Step 3 of 5: Checking ioBroker installation              -----
--------------------------------------------------------------------------------
 
(Re)setting permissions (This might take a while! Please be patient!)... Done.
 
Fixing "sudo-bug" by replacing sudo with gosu... Done.
 
There was a problem detecting the admin instance of your iobroker.
Make sure the ioBroker installation you use has an admin instance or start over with a fresh installation and restore your configuration.
For more details see https://docs.buanet.de/iobroker-docker-image/docs/#restore
 
[DEBUG] Debug mode prevents the container from exiting on errors.
[DEBUG] This enables you to investigate or fix your issue on the command line.
[DEBUG] If you want to stop or restart your container you have to do it manually.
[DEBUG] IoBroker is not running!
```

**Comment by buanet** [comment]
Hallo,
zunächst einmal scheint dein Docker Container keine Internetverbindung oder zumindest Probleme mit dem DNS zu haben:
```
Updating Linux packages on first run...
W: Failed to fetch http://deb.debian.org/debian/dists/bullseye/InRelease Temporary failure resolving 'deb.debian.org'
```
Dann scheint dein mount zu funktionieren, denn er erkennt die iobroker Installation im Verzeichnis:
```
Existing installation of ioBroker detected in "/opt/iobroker".
```
Am Ende sieht es so aus als gibt es ein problem den iobroker befehl aufzurufen. 
Verbinde dich mal auf die Kommandozeile in den Container und mache ein `iob status` un poste die Antwort. 

MfG,
André

**Reply by IrgendwerUsingGitHub** [reply]
Hallo,

Danke für deine Antwort und ich find es mega dass du iobroker in docker umgesetzt hast.

So. Also Internetverbindung. Ich hab sowohl 8.8.8.8 als auch Netzwerk intern die fritzbox erfolgreich angepingt

iob status:
`root@5b16d513a98b:/opt/iobroker# iob status
No connection to databases possible ...`

Edit:
DNS funktioniert nicht. ping auf www.google.de geht nicht
Fritzbox neu gestartet und jetzt funktionert zumindest der DNS. Ansonsten bleibt das Problem aber bestehen

**Comment by buanet** [comment, marked as answer]
Jo, da hast du deinen Fehler. 
```
No connection to databases possible ...
```
Dein ioBroker ist futsch. Dass er seine Datenbank nicht öffnen kann, kann viele Ursachen haben. Leider weiß ich nicht wie deine Konfig vom Container aussieht, oder was du bereits gemacht hast. Warst du mit Adaptern und js-controller aktuell?

Das hier
> Ich habe versucht dem binding Ordner und sämtlichen Inhalt chmod 777 zu setzen, aber leider erfolglos.

Ist auch keine gute Idee gewesen. Ist aber vermutlich eher nicht die Ursache.  

Aus meiner Sicht hast du zwei Optionen.
1. iobroker forum durcharbeiten und schauen welche ursachen es haben kann, dass dein iobroker keine Verbindung zur DB aufbauen kann
2. einen frischen container mit neuem mount aufsetzen und ein backup zurück spielen

Wenn es bei einem frischen container mit mount schon probleme geben sollte, dann musst du mal checken ob das bei dir mit UID und GID = 1000 hin kommt. Aufgrund der Architektur tippe ich mal auf nen Pi? Da sollte es eigentlich passen. Ist es eine DiskStation, dann sind die IDs tendenziell nicht korrekt gesetzt...  Aber wie gesagt, wenn es ein Rechteproblem sein sollte, hätte ich als Fehlermeldung ein `permission denied` erwartet.

MfG
André
